compiler/optimizer/OMRValuePropagation.cpp

```diff
@@ -1848,6 +1848,15 @@ TR::VPConstraint *OMR::ValuePropagation::mergeDefConstraints(TR::Node *node, int
    {
    isGlobal = true; // Will be reset if local constraints found
 
+   if (_defMergedNodes->get(node->getGlobalIndex()))
```

Comment:
This is just a performance optimization unrelated to the functional bug you are fixing, correct ? i.e. avoiding repeatedly looking at the same store node.

Reply:
_defMergedNodes seems to be initialized on a "per structure" basis. Will this work as you intend for a nested loop for example ? i.e. are we sure we won't end up having the processing of the outer loop and inner loop interfere in some way ?

Reply:
_defMergedNodes is not an optimization it is required for correctness - it prevents us from merging backedge constraints for a value has already been evaluated and at the point of first evaluation was unconstrained resulting in us storing a NULL. The processing won't interfere because it is an evaluation order check done locally.